GW2EIEvtcParser/EncounterLogic/Raids/W3/TwistedCastle.cs

```diff
@@ -113,5 +114,52 @@ internal override string GetLogicName(CombatData combatData, AgentData agentData
         {
             return "Twisted Castle";
         }
+
+        protected override void SetInstanceBuffs(ParsedEvtcLog log)
+        {
+            base.SetInstanceBuffs(log);
+            IReadOnlyList<AbstractBuffEvent> mildlyInsane = log.CombatData.GetBuffData(AchievementEligibilityMildlyInsane);
+            bool hasBeenAdded = false;
+
+            if (log.FightData.Success)
+            {
+                if (mildlyInsane.Any())
+                {
+                    foreach (Player p in log.PlayerList)
+                    {
+                        if (p.HasBuff(log, AchievementEligibilityMildlyInsane, log.FightData.FightEnd - ServerDelayConstant))
+                        {
+                            InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityMildlyInsane], 1));
+                            hasBeenAdded = true;
+                            break;
+                        }
+                    }
+                }
+                if (CustomCheckMildlyInsaneEligibility(log) && !hasBeenAdded)
+                {
+                    InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityMildlyInsane], 1));
+                }
+            }
+        }
+
+        private static bool CustomCheckMildlyInsaneEligibility(ParsedEvtcLog log)
+        {
+            foreach (Player player in log.PlayerList)
+            {
+                IReadOnlyDictionary<long, BuffsGraphModel> bgms = player.GetBuffGraphs(log);
+                if (bgms != null && bgms.ContainsKey(Madness))
```

Comment:
ContainsKey to be replaced by TryGetValue(Madness, out BuffsGraphModel bgm), remove the one below